app/services/fact_cleaner.rb

```diff
@@ -1,80 +1,54 @@
 class FactCleaner
   delegate :logger, :to => :Rails
 
-  def clean!
-    @deleted_count = delete_excluded_facts + delete_orphaned_facts
+  def clean!(batch_size = 500)
```

Comment:
We can pass the batch size in the constructor and save it as instance variable.
It will make all method signatures cleaner: 
```suggestion
  def initialize(batch_size: 500)
    @batch_size = batch_size
  end
  
  def clean!
```